launcher/icons/IconList.cpp

```diff
@@ -77,10 +77,90 @@ IconList::IconList(const QStringList& builtinPaths, QString path, QObject* paren
 void IconList::sortIconList()
 {
     qDebug() << "Sorting icon list...";
-    std::sort(icons.begin(), icons.end(), [](const MMCIcon& a, const MMCIcon& b) { return a.m_key.localeAwareCompare(b.m_key) < 0; });
+    std::sort(icons.begin(), icons.end(), [](const MMCIcon& a, const MMCIcon& b) {
+        bool aIsSubdir = a.m_key.contains(QDir::separator());
+        bool bIsSubdir = b.m_key.contains(QDir::separator());
+        if (aIsSubdir != bIsSubdir) {
+            return !aIsSubdir;  // root-level icons come first
+        }
+        return a.m_key.localeAwareCompare(b.m_key) < 0;
+    });
     reindex();
 }
 
+// Helper function to add directories recursively
+bool IconList::addPathRecursively(const QString& path)
+{
+    QDir dir(path);
+    if (!dir.exists())
+        return false;
+
+    bool watching = false;
+
+    // Add the directory itself
+    watching = m_watcher->addPath(path);
+
+    // Add all subdirectories
+    QFileInfoList entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+    for (const QFileInfo& entry : entries) {
+        if (addPathRecursively(entry.absoluteFilePath())) {
+            watching = true;
+        }
```

Comment:
You can write the if as single or operation(not sure if this is more readable or not):
```suggestion
        watching |= addPathRecursively(entry.absoluteFilePath());
```

Reply:
It doesn't look clearer imo